Refactored a test intended to examine different clocks.

AWSHurneyt opened the pull request:
### Description
The [getNextExecutionTime ](https://github.com/opensearch-project/job-scheduler/blob/main/spi/src/main/java/org/opensearch/jobscheduler/spi/schedule/CronSchedule.java#L86) method only calls the configured clock when the `instant` parameter is `null`. As a result, `getNextExecutionTime` would use the `now` `instant` with the default `zoneId` for both cron expressions. Refactored the test to use an `instant` with a set epoch, and `zoneId`s specific to the test cases.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

lezzago commented on spi/src/test/java/org/opensearch/jobscheduler/spi/schedule/CronScheduleTests.java:
```diff
@@ -38,16 +38,12 @@ public void setup() {
         assertEquals("Next execution time based on different clock should be same.",
-            pdtClockCronSchedule.getNextExecutionTime(now),
-            utcClockCronSchedule.getNextExecutionTime(now));
+            pdtClockCronSchedule.getNextExecutionTime(instant.atZone(ZoneId.of("America/Los_Angeles")).toInstant()),
+            utcClockCronSchedule.getNextExecutionTime(instant.atZone(ZoneId.of("UTC")).toInstant()));
```
I prefer the old test. This seems like it only tests if `instant.atZone(ZoneId.of("America/Los_Angeles")).toInstant()` is equal to `instant.atZone(ZoneId.of("UTC")).toInstant())` which is not what we are trying to test. This modification makes both `pdtClockCronSchedule` and `utcClockCronSchedule` as they are using the same timezone.

AWSHurneyt replied:
Good point. Refactored the test.

downsrob commented:
I can see this being potentially flaky but I doubt that it will be a big problem with ms precision. If this test does cause flaky failures in the future, let's either remove it or find a way to mock the clocks to get rid of the flakiness.

AWSHurneyt commented:
> I can see this being potentially flaky but I doubt that it will be a big problem with ms precision. If this test does cause flaky failures in the future, let's either remove it or find a way to mock the clocks to get rid of the flakiness.

Rather than mock the clock, I refactored the test to make use of `Clock.fixed()`.

lezzago commented on spi/src/test/java/org/opensearch/jobscheduler/spi/schedule/CronScheduleTests.java:
```diff
@@ -38,16 +38,16 @@ public void setup() {
-        Clock pdtClock = Clock.system(ZoneId.of("America/Los_Angeles"));
-        Clock utcClock = Clock.system(ZoneId.of("UTC"));
-        CronSchedule pdtClockCronSchedule = new CronSchedule("* * * * *", ZoneId.systemDefault());
+        Instant now = Instant.now();
```
This seems like its not being used
